OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/ResetPasswordAlertDialogFragment.java

```diff
@@ -0,0 +1,190 @@
+package org.edx.mobile.view.dialog;
+
+import android.app.Activity;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.design.widget.TextInputEditText;
+import android.support.design.widget.TextInputLayout;
+import android.support.v7.app.AlertDialog;
+import android.view.View;
+import android.view.accessibility.AccessibilityEvent;
+
+import com.google.inject.Inject;
+
+import org.edx.mobile.R;
+import org.edx.mobile.authentication.LoginService;
+import org.edx.mobile.http.CallTrigger;
+import org.edx.mobile.http.ErrorHandlingCallback;
+import org.edx.mobile.interfaces.OnActivityResultListener;
+import org.edx.mobile.model.api.ResetPasswordResponse;
+import org.edx.mobile.util.InputValidationUtil;
+import org.edx.mobile.util.NetworkUtil;
+import org.edx.mobile.util.images.ErrorUtils;
+
+public class ResetPasswordAlertDialogFragment extends AlertDialogFragment {
+    @Inject
+    private LoginService loginService;
+
+    protected static final String EXTRA_LOGIN_EMAIL = "login_email";
+
+    private TextInputEditText editText;
+    private TextInputLayout textInputLayout;
+    private OnActivityResultListener listener;
+
+    public static ResetPasswordAlertDialogFragment newInstance(@NonNull Context context, @Nullable String email) {
+        ResetPasswordAlertDialogFragment fragment = new ResetPasswordAlertDialogFragment();
+        Bundle args = new Bundle();
+        args.putString(ARG_TITLE, context.getResources().getString(R.string.confirm_dialog_title_help));
+        args.putString(ARG_MESSAGE, context.getResources().getString(R.string.confirm_dialog_message_help));
+        args.putString(EXTRA_LOGIN_EMAIL, email);
+        fragment.setArguments(args);
+
+        if (context instanceof OnActivityResultListener) {
+            fragment.setListener((OnActivityResultListener) context);
+        } else {
+            throw new ClassCastException("Caller should implement the OnActivityResultListener interface.");
+        }
+        return fragment;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        AlertDialog alertDialog = (AlertDialog) super.onCreateDialog(savedInstanceState);
+
+        editText = new TextInputEditText(getContext());
+        textInputLayout = new TextInputLayout(getContext());
+        textInputLayout.addView(editText);
+        textInputLayout.setHint(getContext().getString(R.string.email));
+
+        int horizontalPadding = Math.round(getResources().getDimension(R.dimen.abc_dialog_padding_material));
```

Comment:
Note that you should be getting this value from the `dialogPreferredPadding` theme attribute instead of a hardcoded specific dimension.